CONTRIBUTING.md

```diff
@@ -1,6 +1,29 @@
-Before contributing code, please note:
+# Before reporting a bug
+1. Search the [issue tracker](https://github.com/aframevr/aframe/issues) for similar issues. 
+2. Try the latest dev branch version of [aframe](https://github.com/aframevr/aframe/tree/dev).
+
+# How to report a bug
+1. Specify the revision number of the aframe.js library where the bug occurred.
+2. If you are using a PC, specify your browser version, operating system, and graphics card (for example, Chrome 23.0.1271.95, Windows 7, Nvidia Quadro 2000M).
+3. If you are using a mobile device, specify your make, model, operating system version, and browser (for example, Samsung Galaxy Nexus, Android 4.4.4, Chrome 47.0.2526.83).
+3. Describe the problem in detail (i.e., what happened and what you expected would happen).
+4. Provide a small test-case (http://codepen.io). [Here is a pen](http://codepen.io/anon/pen/KVWpbb) that uses the dev branch. If a test-case is not possible, provide a link to a live version of your application.
+5. If helpful, include a screenshot. Annotate the screenshot for clarity.
+
+# How to join us on slack
+1. Request an invitation from our [automated system](https://aframevr-slack.herokuapp.com/)
+2. [Join](https://aframevr.slack.com)  the discussion
+
+# How to contribute to aframe
+1. Make sure you have a [GitHub account](https://github.com/join).
+2. [Fork](https://github.com/aframevr/aframe/fork) the repository on GitHub.
+3. Checkout the [dev branch](https://github.com/aframevr/aframe/tree/dev).
+4. Make changes to your fork of the repository.
+5. Submit a pull request.
+
+### Before contributing code, please note:
 
 * You agree to license your contributions under the [license](LICENSE).
 * Follow [jshint](http://eslint.org/docs/rules/) [see our rules](.jshintrc).
```

Comment:
@ngokevin: Can you elaborate on this? Should I remove the item about jshint as this is taken care of automatically?